components/camel-aws/src/main/java/org/apache/camel/component/aws/sns/SnsEndpoint.java

```diff
@@ -92,6 +107,11 @@ public void doStart() throws Exception {
             LOG.trace("Updating the SNS region with : {} " + configuration.getAmazonSNSEndpoint());
             snsClient.setEndpoint(configuration.getAmazonSNSEndpoint());
         }
+
+        // check the setting the headerFilterStrategy
+        if (headerFilterStrategy == null) {
+            headerFilterStrategy = new SnsHeaderFilterStrategy();
```

Comment:
That might be true, but I just copied it from the SqsEndpoint implementation. As already said, I think it should be refactored, so the duplicate code is avoided, but I did not want to make the change too big.